gematria/experiments/access_pattern_bm/contiguous_matrix.cc

```diff
@@ -38,16 +39,18 @@ std::unique_ptr<int[]> CreateRandomContiguousMatrix(const std::size_t size) {
   return matrix;
 }
 
-void FlushContiguousMatrixFromCache(std::unique_ptr<int[]> &matrix,
+// Flushes matrices created by `CreateRandomContiguousMatrix`. Takes the matrix
+// itself, along with its size (given that it is a size x size square matrix,
+// holding size * size ints).
+void FlushContiguousMatrixFromCache(const void *matrix,
                                     const std::size_t size) {
   constexpr int line_size = ABSL_CACHELINE_SIZE;
-  const char *ptr = (const char *)matrix.get();
-  const char *end = (const char *)(matrix.get() + (size + 1) * sizeof(int));
+  const char *begin = reinterpret_cast<const char *>(matrix);
+  const char *end = begin + size * size * sizeof(int) / sizeof(char);
```

Comment:
`sizeof(char)` is always one by definition. Which might sound a bit weird, but that's what the spec says: https://en.cppreference.com/w/cpp/language/sizeof